third_party/pyth/multisig-wh-message-builder/src/index.ts

```diff
@@ -58,6 +72,84 @@ program
   .description("CLI to creating and executing multisig transactions for pyth")
   .version("0.1.0");
 
+program
+  .command("init-vault")
+  .description(
+    "Initialize a new multisig vault. NOTE: It's unlikely that you need run this manually. Primarily used in the Tilt local devnet"
+  )
+  .requiredOption(
+    "-k --create-key <address>",
+    "Vault create key. It's a pubkey used to seed the vault's address"
+  )
+  .requiredOption(
+    "-m --mesh-program <address>",
+    "Mesh multisig program address"
+  )
+  .requiredOption(
+    "-x --external-authority <address>",
+    "External authority address"
+  )
+  .option("-c --cluster <network>", "solana cluster to use", "devnet")
+  .option("-p --payer <filepath>", "payer keypair file")
+  .option(
+    "-t --threshold <threshold_number>",
+    "Approval quorum threshold for the vault",
+    "2"
+  )
+  .requiredOption(
+    "-i --initial-members <comma_separated_members>",
+    "comma-separated list of initial multisig members, without spaces"
+  )
+  .option("-r --solana-rpc <url>", "Solana RPC address to use")
+  .action(async (options: any) => {
+    let cluster: Cluster = options.cluster;
+    let createKeyAddr: PublicKey = new PublicKey(options.createKey);
+    let extAuthorityAddr: PublicKey = new PublicKey(options.externalAuthority);
+    let meshProgramAddr: PublicKey = new PublicKey(options.meshProgram);
+
+    let threshold: number = parseInt(options.threshold, 10);
+
+    let initialMembers = options.initialMembers
+      .split(",")
+      .map((m: string) => new PublicKey(m));
+
+    let mesh = await getSquadsClient(
+      cluster,
+      options.ledger,
+      options.ledgerDerivationAccount,
+      options.ledgerDerivationChange,
+      options.payer,
+      cluster == "localdevnet" ? options.solanaRpc : undefined,
+      cluster == "localdevnet" ? meshProgramAddr : undefined
+    );
+
+    // The vault key is a PDA and the create key is only an ingredient for it
+    let [vaultAddr, bump] = await PublicKey.findProgramAddress(
```

Comment:
For this I think you can use `getMsPDA` method from squads (in the lib, not in the class). They have some PDA methods [here](https://github.com/Squads-Protocol/mesh-sdk/blob/main/src/address.ts)

Reply:
Amazing, thank you!